### PR TITLE
Add `build` subcommand for building project

### PR DESCRIPTION
**Pull request by snideto**
Move build functionality from grid-tied.

### CODE REVIEW

File: .project [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+		<buildCommand>
+			<name>org.python.pydev.PyDevBuilder</name>
+			<arguments>
+			</arguments>
+		</buildCommand>
+	</buildSpec>
+	<natures>
+		<nature>org.python.pydev.pythonNature</nature>
+	</natures>
+</projectDescription>
```
Comment (altendky) [inline .project:17]: I'll let adding these slide.  :]

**Review by altendky — APPROVED** [review]

**Review by altendky — CHANGES_REQUESTED** [review]
`build_type_choices` probably belongs in `api` and should become an enumeration now that it can be selected by something other than CLI text.

**Comment by snideto** [comment]
https://github.com/pallets/click/issues/605

**Comment by snideto** [comment]
github.com/epcpower/ccstudiodss/commit/0641a7d5924b5ba1686f6b7f70ec881b1b6e998d

**Comment by altendky** [comment]
Sorry, I see the click link now.  Here's the basic conversion but I'll gander through the link for something niftier. Maybe an actual click type.

```python3
build_type_strings = [build_type.name for build_type in Build types]

build_type = BuildTypes[chosen_string]
```

**Comment by altendky** [comment]
Yeah, no named tuple.  Stdlib Enum or enum34 from PyPI.  Either PR for Click similar to the last suggestion in the ticket or something like I have above.  I haven't contributed to click so I don't know how the process goes with them.

**Review by altendky — APPROVED** [review]

**Comment by altendky** [comment]
Note that GitHub inserts the commits into the history here so no need to link them.

**Pull request merged**